store/storage/storage_test_suite.go

```diff
@@ -122,6 +122,30 @@ func (s *StorageTestSuite) TestDatabase_GetVersionedKey() {
 	ok, err = db.Has(storeKey1, 10, []byte("key"))
 	s.Require().NoError(err)
 	s.Require().True(ok)
+
+	for i := uint64(11); i <= 14; i++ {
+		bz, err = db.Get(storeKey1, i, []byte("key"))
+		s.Require().NoError(err)
+		s.Require().Equal([]byte("value011"), bz)
+
+		ok, err = db.Has(storeKey1, i, []byte("key"))
+		s.Require().NoError(err)
+		s.Require().True(ok)
+	}
+
+	err = db.Delete(storeKey1, 15, []byte("key"))
+	s.Require().NoError(err)
+
+	for i := uint64(15); i <= 17; i++ {
+		bz, err = db.Get(storeKey1, i, []byte("key"))
+		s.Require().NoError(err)
+		// TODO should we assert nil?
+		s.Require().Empty(bz)
```

Comment:
If it does not exist, yeah I'd assert nil

Reply:
but right now, the implementation will fail, because it returns empty slice.

Reply:
for which backend?

Reply:
can't reproduce now, not sure what happended.

Reply:
```
    storage_test_suite.go:142:
        	Error Trace:	/Users/yihuang/src/cosmos-sdk/store/storage/storage_test_suite.go:142
        	Error:      	Expected nil, but got: []byte{0x76, 0x61, 0x6c, 0x75, 0x65, 0x30, 0x31, 0x31}
        	Test:       	TestStorageTestSuite/TestDatabase_GetVersionedKey
```

only that pebbledb backend fail for this case.

Reply:
```
    storage_test_suite.go:149:
        	Error Trace:	/Users/yihuang/src/cosmos-sdk/store/storage/storage_test_suite.go:149
        	Error:      	Expected nil, but got: []byte{0x76, 0x61, 0x6c, 0x75, 0x65, 0x30, 0x31, 0x31}
        	Test:       	TestStorageTestSuite/TestDatabase_GetVersionedKey
```

sqlite backend fail this case as well.

Reply:
Meaning, they don't delete the value when we expect them to?

Reply:
or the query don't find out it's deleted.